EnvironmentSimulator/Modules/RoadManager/RoadManager.cpp

```diff
@@ -1685,6 +1712,50 @@ bool Road::GetZAndPitchByS(double s, double *z, double *pitch, int *index)
 	return false;
 }
 
+bool Road::UpdateZAndRollBySAndT(double s, double t, double *z, double *roll, int *index)
+{
+	if (GetNumberOfSuperElevations() > 0)
+	{
+		if (*index < 0 || *index >= GetNumberOfSuperElevations())
+		{
+			*index = 0;
+		}
+		Elevation *super_elevation = GetSuperElevation(*index);
+		if (super_elevation == NULL)
+		{
+			LOG("Superelevation error NULL, nelev: %d elev_idx: %d\n", GetNumberOfSuperElevations(), *index);
+			return false;
+		}
+
+		if (super_elevation && s > super_elevation->GetS() + super_elevation->GetLength())
+		{
+			while (s > super_elevation->GetS() + super_elevation->GetLength() && *index < GetNumberOfSuperElevations() - 1)
+			{
+				// Move to next elevation section
+				super_elevation = GetSuperElevation(++ *index);
+			}
+		}
+		else if (super_elevation && s < super_elevation->GetS())
+		{
+			while (s < super_elevation->GetS() && *index > 0)
+			{
+				// Move to previous elevation section
+				super_elevation = GetSuperElevation(-- *index);
```

Comment:
I see. The superelevation is a polynomial along s, and roll is constant in t dimension. I could expected support for linear change along s, but I checked the standard and the implementation is correct, only poly3 is used. Which makes sense since it can represent line as well. Good.